linera-rpc/src/grpc/transport.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::NodeOptions;
+
+#[derive(Clone, Debug, Default)]
+pub struct Options {
+    pub connect_timeout: Option<std::time::Duration>,
+    pub timeout: Option<std::time::Duration>,
+}
+
+impl From<&'_ NodeOptions> for Options {
+    fn from(node_options: &NodeOptions) -> Self {
+        Self {
+            connect_timeout: Some(node_options.send_timeout),
+            timeout: Some(node_options.recv_timeout),
+        }
+    }
+}
+
+#[cfg(web)]
+mod implementation {
+    use super::Options;
+
+    pub use tonic_web_wasm_client::{Client as Channel, Error};
+
+    pub fn create_channel(address: String, _options: &Options) -> Result<Channel, super::Error> {
+        // TODO this should respect `options`
+        Ok(tonic_web_wasm_client::Client::new(address))
+    }
+}
+
+#[cfg(not(web))]
+mod implementation {
+    use super::Options;
+
+    pub use tonic::transport::{Channel, Error};
+
+    pub fn create_channel(
+        address: String,
+        options: &Options,
+    ) -> Result<Channel, super::super::GrpcError> {
```

Comment:
Minor: could this be `crate::GrpcError` instead (or preferably importing `GrpcError` at the start of the module)?

```suggestion
    ) -> Result<Channel, crate::GrpcError> {
```

Reply:
Oops, counted parent modules wrong :sweat_smile: I would still prefer to `use super::super::GrpcError`, but this is not a blocker.

Reply:
I think I prefer to keep this import module-relative because it means we can move the whole `grpc` module around without having to rewrite it.

If I referred to it more than once I'd definitely `use` it, but I'm not sure about the benefit when it's only ever referred to once. `use`s add a layer of indirection that makes it less clear where the name comes from, so if I'm to `use` something I usually want to get some benefit out of it in terms of lines of code (with an exception for things like `async_trait` or certain names like `Future`, which are sufficiently well-known that even unqualified it's always obvious to an experienced Rust programmer what they mean).

Reply:
I've added `cfg-if` as a dependency and used it to factor away the whole implementation-module noise :)